ArdusatSDK.h

```diff
@@ -315,10 +315,13 @@ class Orientation: public Sensor {
 
     boolean read(Acceleration & accel, Magnetic & mag);
 
-    const char * toCSV(const char * sensorName);
-    const char * toJSON(const char * sensorName);
+    const char * readToCSV(const char * sensorName);
```

Comment:
ya, adding `virtual` means that when methods get called, they are dispatched to something called the virtual lookup table (http://www.learncpp.com/cpp-tutorial/125-the-virtual-table/) which is necessary for polymorphism to work correctly. This is important for the following reason:

No `virtual` declaration:
```cpp
class SuperClass {
  void msg(void) {
    stdout << "parent method" << endl;
  } 
};

class DerivedClass {
  void msg(void) {
    stdout << "derived method" << endl;
  }
};

int main() {
  DerivedClass derived;
  SuperClass *super = &derived;
  
  derived.msg();
  super->msg();
}
```

With classes defined as such, we'll get the following output:

```
derived method
parent method
``` 

But from a polymorphic perspective, we would expect to get

```
derived method
derived method
```

The reason for this is that without the `virtual` declaration C++ doesn't dispatch to the lookup table when you call the method, it just goes straight for the memory address that it expects for that method based on what it thinks the class is. The fact that the two methods are named the same doesn't really bother it.